### PR TITLE
feat(FormContext): Add context for Form to manage input state

### LINKED ISSUE #7507: FormContext - Add context for Form to manage input state
**Is this a new component or an extension of an existing one?**
This is a new component for providing Context to a Form, FormContextProvider.

**Describe the feature**
As a developer, I would like to be able to track input values, errors, validation, and what is touched in a Form by offering a central location for this state by way of context.

**Any other information?**
- This context will be usable independently of the Form component for now, but the intention is to also modify Form (Wrap Form in a FormContextProvider) and FormGroup to utilize this context so that setters and getters won't have to be called manually.
- This is the first step towards a home-grown solution to form state management for Patternfly.

### PR DESCRIPTION
**Pull request by jpuzz0**
**What**: Closes #7507

### CODE REVIEW

File: packages/react-core/src/components/EmptyState/__tests__/Generated/__snapshots__/EmptyStateIcon.test.tsx.snap [hunk 1]
```diff
@@ -10,7 +10,7 @@ exports[`EmptyStateIcon should match snapshot (auto-generated) 1`] = `
     height="1em"
     role="img"
     style="vertical-align: -0.125em;"
-    viewBox="0 0 830 1024"
+    viewBox="0 0 448 512"
```
Comment (jeffpuzzo) [inline packages/react-core/src/components/EmptyState/__tests__/Generated/__snapshots__/EmptyStateIcon.test.tsx.snap:13]: When rebasing with the v5 branch, re-installing node modules and re-running tests, this snapshot was updated, which I would have expected to have been changed in a prior PR.

File: packages/react-core/src/components/TreeView/__tests__/__snapshots__/TreeView.test.tsx.snap [hunk 2]
```diff
@@ -2309,11 +2309,11 @@ exports[`tree view renders icons successfully 1`] = `
                   height="1em"
                   role="img"
                   style="vertical-align: -0.125em;"
-                  viewBox="0 0 1217 1024"
+                  viewBox="0 0 576 512"
```
Comment (jeffpuzzo) [inline packages/react-core/src/components/TreeView/__tests__/__snapshots__/TreeView.test.tsx.snap:2312]: When rebasing with the v5 branch, re-installing node modules and re-running tests, this snapshot was updated, which I would have expected to have been changed in a prior PR.

File: packages/react-core/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
   "homepage": "https://github.com/patternfly/patternfly-react#readme",
   "scripts": {
     "build:umd": "rollup -c --environment IS_PRODUCTION",
-    "clean": "rimraf dist",
+    "clean": "rimraf dist components layouts helpers next deprecated node_modules",
```
Comment (jeffpuzzo) [inline packages/react-core/package.json:43]: This does not pertain to this exact change, but something noticed when running `clean`, I believe these built sub-directories should be included.

File: packages/react-docs/package.json [hunk 4]
```diff
@@ -15,7 +15,7 @@
   "scripts": {
     "analyze": "yarn build:docs --analyze",
     "build:docs": "pf-docs-framework build all",
-    "clean": "rimraf .cache public static/assets static/base.css src/generated/**/*.js",
+    "clean": "rimraf .cache public static/assets static/base.css src/generated/**/*.js node_modules",
```
Comment (jeffpuzzo) [inline packages/react-docs/package.json:18]: When using a local version of the documentation-framework, it helps to clear out these node_modules, as sometimes not doing so leads to false-positives/negatives.

File: packages/react-core/src/components/Form/examples/Form.md [hunk 5]
```diff
@@ -13,8 +13,10 @@ propComponents:
     'FormFieldGroupExpandable',
     'FormFieldGroupHeader',
     'FormFieldGroupHeaderTitleTextObject',
+    'FormContextProps',
+    'FormContextProviderProps',
```
Comment (jeffpuzzo) [inline packages/react-core/src/components/Form/examples/Form.md:17]: Both of these types should probably be marked as beta, but there is no way to do so yet with the PF documentation-framework. The only use-case is an example called `FormState`, which is marked as `beta`, so hopefully that will suffice for now.

File: packages/react-core/src/components/Form/__tests__/FormContext.test.tsx [hunk 6]
```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { FormContextConsumer, FormContextProvider } from '../FormContext';
+
+const customRender = (ui, initialValues?: Record<string, any>) => {
+  return render(<FormContextProvider initialValues={initialValues}>{ui}</FormContextProvider>);
+};
+
+describe('FormContext', () => {
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/Form/__tests__/FormContext.test.tsx:10]: Nit: we're trying to [standardize](https://github.com/patternfly/patternfly-react/wiki/React-Testing-Library-Basics,-Best-Practices,-and-Guidelines#test-nesting-and-describe-block-usage) on not just grouping whole test files in describe blocks

File: packages/react-core/src/components/Form/__tests__/FormContext.test.tsx [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+          </>
+        )}
+      </FormContextConsumer>,
+      {
+        someId: 'some value'
+      }
+    );
+
+    expect(screen.getByText('some value')).toBeVisible();
+  });
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/Form/__tests__/FormContext.test.tsx:78]: I might just be misunderstanding this test, but shouldn't the input need to be changed?

File: packages/react-core/src/components/Form/FormContext.tsx [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+  /** Set a validator for a specific fieldId */
+  setValidator(fieldId: string, validate: (value: string) => string | null): void;
+}
+
+const FormContext = React.createContext({} as FormContextProps);
+
+export const FormContextConsumer = FormContext.Consumer;
+
+export interface FormContextProviderProps {
+  /** Record of intitial values */
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/Form/FormContext.tsx:43]: Nit
```suggestion
  /** Record of initial values */
```

**Review by wise-king-sullyman — COMMENTED** [review]
Neat! I just have a couple small nit things and one test I'm not sure about.

File: packages/react-core/src/components/Form/__tests__/FormContext.test.tsx (hunk 6 above)
Comment (jpuzz0) [reply packages/react-core/src/components/Form/__tests__/FormContext.test.tsx:10]: Yeah, I wrote this back before we started that standard, I should have caught that before opening this. Will update.

File: packages/react-core/src/components/Form/__tests__/FormContext.test.tsx (hunk 7 above)
Comment (jpuzz0) [reply packages/react-core/src/components/Form/__tests__/FormContext.test.tsx:78]: I think that was the intention, yes. It looks like the test just below this does exactly that, so I'll just remove this one.

**Review by wise-king-sullyman — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

**Comment by nicolethoen** [comment]
Github wont let me merge this because it says there are still conflicts needing to be resolved?

**Comment by jpuzz0** [comment]
> Github wont let me merge this because it says there are still conflicts needing to be resolved?

Yeah, looks like another PR merged today led to another conflict. I rebased and resolved that conflict just now.

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**